framework/src/transfers/MultiAppVariableValueSamplePostprocessorTransfer.C

```diff
@@ -41,15 +47,104 @@ MultiAppVariableValueSamplePostprocessorTransfer::validParams()
 
-  if (isParamValid("from_multi_app"))
-    paramError("from_multi_app", "This transfer direction has not been implemented");
+  if (_directions.contains("from_multiapp"))
+  {
+    // Check that the variable is a CONSTANT MONOMIAL.
+    auto & fe_type = _var.feType();
+    if (fe_type.order != CONSTANT || fe_type.family != MONOMIAL)
+      paramError("source_variable",
+                 "Variable must be in CONSTANT MONOMIAL when transferring from a postprocessor "
+                 "from sub-apps.");
+
+    if (!_fe_problem.getAuxiliarySystem().hasVariable(_var_name))
+      paramError("source_variable", "Variable must be an auxiliary variable");
+  }
+}
+
+void
+MultiAppVariableValueSamplePostprocessorTransfer::setupPostprocessorCommunication()
+{
+  if (!_directions.contains("from_multiapp"))
+    return;
+
+  const auto num_global_apps = getFromMultiApp()->numGlobalApps();
+
+  // Setup the communication pattern
+  _postprocessor_to_processor_id.resize(num_global_apps,
+                                        std::numeric_limits<processor_id_type>::max());
+  for (const auto i : make_range(num_global_apps))
+    if (getFromMultiApp()->hasLocalApp(i))
+      _postprocessor_to_processor_id[i] = this->processor_id();
+
+  _communicator.min(_postprocessor_to_processor_id);
+#ifdef DEBUG
+  for (const auto i : make_range(num_global_apps))
+  {
+    mooseAssert(_postprocessor_to_processor_id[i] != std::numeric_limits<processor_id_type>::max(),
+                "Every element in the vector should have been set.");
+    if (getFromMultiApp()->hasLocalApp(i))
+      mooseAssert(_postprocessor_to_processor_id[i] == this->processor_id(),
+                  "If I owned this app, then the processor id value should be my own");
+  }
+#endif
+}
+
+void
+MultiAppVariableValueSamplePostprocessorTransfer::cacheElemToPostprocessorData()
+{
+  if (!_directions.contains("from_multiapp"))
+    return;
+
+  // Cache the Multiapp position ID for every element.
+  auto & mesh = _fe_problem.mesh().getMesh();
+  unsigned int multiapp_pos_id = 0;
+  for (auto & elem : as_range(mesh.active_local_elements_begin(), mesh.active_local_elements_end()))
+    // Exclude the elements without dofs.
+    if (_var.hasBlocks(elem->subdomain_id()))
+    {
+      Real distance = std::numeric_limits<Real>::max();
+      unsigned int count = 0;
+      for (unsigned int j = 0; j < getFromMultiApp()->numGlobalApps(); ++j)
+      {
+        Real current_distance = (getFromMultiApp()->position(j) - elem->true_centroid()).norm();
+        if (MooseUtils::absoluteFuzzyLessThan(current_distance, distance))
+        {
+          distance = current_distance;
+          multiapp_pos_id = j;
+          count = 0;
+        }
+        else if (MooseUtils::absoluteFuzzyEqual(current_distance, distance))
+          ++count;
+      }
+      if (count > 0)
+        mooseError("The distances of an element to more than one sub-applications are too close."
```

Comment:
```suggestion
        mooseWarning("The distances of an element to more than one sub-applications are too close."
```

Reply:
I prefer using mooseError more often than mooseWarning. Undetermined behavior should be considered as an error. Do you have arguments on making this as a warning?

Reply:
it's not undetermined, you pick the element with the lowest ID. 

The argument is this needs to work for rapid prototyping and sometimes we dont care about this scenario as much as getting something done

Reply:
Not a good argument. I'd rather the users be aware potential issues with more time on fixing the models and instead later complaining the code is not working as expected. The error message should give them enough information to check their models.

Reply:
A warning gives them enough information.
If they dont want to heed it, that's their problem.

An error here breaks this transfer for entire classes of simulations.
Imagine you want to use a centroid multiapp but transfer onto a nodal variable?
Or you run a complex simulation with adaptivity and it develops this condition after 10k CPU hours?
Or the transition from one subapp to another is actually the correct behavior as the mesh is changing?

Reply:
`transition from one subapp to another` won't hit this error. If starting with a CentroidMultiApp, mesh adaptation won't cause troubles. But I will make this as a warning.

Reply:
it wont hit unless the middle point, the one that is undetermined, is on the way. 
CentroidMultiApp would hit for all nodal variables on a regular mesh

ok thx